drivers/spi/spi_numaker.c

```diff
@@ -137,80 +141,58 @@ static int spi_numaker_configure(const struct device *dev, const struct spi_conf
 	return 0;
 }
 
-static int spi_numaker_txrx(const struct device *dev)
+static int spi_numaker_txrx(const struct device *dev, uint8_t spi_dfs)
 {
 	struct spi_numaker_data *data = dev->data;
 	const struct spi_numaker_config *dev_cfg = dev->config;
 	struct spi_context *ctx = &data->ctx;
 	uint32_t tx_frame, rx_frame;
-	uint8_t word_size, spi_dfs;
 	uint32_t time_out_cnt;
 
 	LOG_DBG("%s", __func__);
-	word_size = SPI_WORD_SIZE_GET(ctx->config->operation);
-
-	switch (word_size) {
-	case 8:
-		spi_dfs = 1;
-		break;
-	case 16:
-		spi_dfs = 2;
-		break;
-	case 24:
-		spi_dfs = 3;
-		break;
-	case 32:
-		spi_dfs = 4;
-		break;
-	default:
-		spi_dfs = 0;
-		LOG_ERR("Not support SPI WORD size as [%d] bits", word_size);
-		return -EIO;
-	}
-
-	LOG_DBG("%s -->word_size [%d]", __func__, word_size);
 
 	if (spi_context_tx_on(ctx)) {
 		tx_frame = ((ctx->tx_buf == NULL) ? SPI_NUMAKER_TX_NOP
-						  : UNALIGNED_GET((uint8_t *)(data->ctx.tx_buf)));
+			   : UNALIGNED_GET((uint32_t *)(data->ctx.tx_buf)));
```

Comment:
drop the outer parenthesis while at it, not needed

Reply:
@fabiobaltieri 
drop  the outer parenthesis as below ?
`: UNALIGNED_GET((uint32_t *)data->ctx.tx_buf));`

Reply:
yeah

Reply:
or even make it a proper `if ()`, that statement is busy enough as it is and spans over three lines anyway

Reply:
Thanks of your advice and done